### ISSUE TITLE
npm deployments failing - stash directory nonexistent

### ISSUE THREAD

**Issue by riggerthegeek** [travis-ci/travis-ci#4671]
**tl;dr** Can't deploy my project to npm via Travis.  I have done this with an auth key generated by v2.11.3

I'm trying to [deploy a new release](https://travis-ci.org/riggerthegeek/steeplejack/jobs/76290725) today of a fairly old project.  The last release was done a few weeks ago with the latest v2 npm key, so I don't believe that's the issue behind why I'm getting unauthorised back from npm.

This is the log I'm seeing
 
    Preparing deploy
    Authenticated with email simon@simonemms.com
    /usr/lib/git-core/git-stash: 186: /usr/lib/git-core/git-stash: cannot create /home/travis/build/riggerthegeek/steeplejack/.git/logs/refs/stash: Directory nonexistent

Indicates we're authenticated, but we're getting a stash error

	NPM API key format changed recently. If your deployment fails, check your API key in ~/.npmrc.
	http://docs.travis-ci.com/user/deployment/npm/
	npm ERR! Linux 3.13.0-40-generic
	npm ERR! argv "/home/travis/.nvm/versions/node/v0.12.7/bin/node" "/home/travis/.nvm/versions/node/v0.12.7/bin/npm" "publish"
	npm ERR! node v0.12.7
	npm ERR! npm  v2.11.3
	npm ERR! code ENEEDAUTH
	npm ERR! need auth auth required for publishing
	npm ERR! need auth You need to authorize this machine using `npm adduser`
	npm ERR! Please include the following file with any support request:
	npm ERR!     /home/travis/build/riggerthegeek/steeplejack/npm-debug.log
	No stash found.

This is not an isolated problem.  I'm also getting it on another [project of mine](https://travis-ci.org/riggerthegeek/steeplejack-errors/jobs/76290555)

Please help

**Comment by BanzaiMan** [comment]
@riggerthegeek Could you double check your NPM API key?

**Comment by riggerthegeek** [comment]
I've logged in three times and changed it.  The string in my .npmrc file is

    //registry.npmjs.org/:_authToken=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

I've tried using the whole line and just the authToken= value.  I'm configuring it using `travis encrypt --add deploy.api_key`.

Have got v1.8.0 of the travis gem installed on Mac OSX Yosemite

**Comment by BanzaiMan** [comment]
@riggerthegeek Thanks for the info. I've identified and fixed the issue. Could you try deploying again?

**Comment by riggerthegeek** [comment]
@BanzaiMan yup, that seems to have fixed it.  Thanks.

I wasn't sure it wasn't me being stupid, especially as I couldn't see anyone else having problems.  Glad I'm not going stupid :)

[Build](https://travis-ci.org/riggerthegeek/steeplejack/jobs/76320678) here for you to see
